src/libcore/ptr/const_ptr.rs

```diff
@@ -295,6 +295,64 @@ impl<T: ?Sized> *const T {
         intrinsics::ptr_offset_from(self, origin)
     }
 
+    /// Returns whether two pointers are guaranteed equal. Returns `true` whenever the
+    /// pointers are known to be equal.
+    /// In some contexts (e.g., compile-time evaluation),
+    /// it is not always possible to determine equality of two pointers, so this function may
+    /// spuriously return false for pointers that later actually turn out to be equal.
+    /// But when it returns true, the pointers are guaranteed to be equal.
```

Comment:
```suggestion
    /// spuriously return `false` for pointers that later actually turn out to be equal.
    /// But when it returns `true`, the pointers are guaranteed to be equal.
```

Same for `guaranteed_ne`.